src/core/blockchain/celo/contracts/election.ts

```diff
@@ -26,28 +28,35 @@ export class Election {
             abi
                 .simpleEncode(
                     'vote(address,uint256,address,address)',
-                    address,
+                    validator.id,
                     new BigNumber(tx.amount).toFixed(),
                     lesser,
                     greater
                 )
                 .toString('hex');
 
-        const fees = await this.client.getFees(TransactionType.CONTRACT_CALL, {
-            from: tx.account.address,
-            to: address,
-            amount: tx.amount,
-            contractAddress,
-            raw
-        });
+        const fees = await this.client.getFees(
+            TransactionType.CONTRACT_CALL,
+            {
+                from: tx.account.address,
+                to: validator.id,
+                amount: tx.amount,
+                contractAddress,
+                raw
+            },
+            TokenType.ERC20
```

Comment:
this is confusing, is it really needed ?

Reply:
If i don't use TokenType.ERC20 then the estimate on gas will take the defaultConfig from native token

Reply:
well, then the estimation is not quite correct, here you use voting smart contract..
add an issue or a comment, we have to fix this. Maybe we need to refactor gas estimation method